server/helpers.js

```diff
@@ -0,0 +1,155 @@
+ * @param {Object|Config} options
+ * @returns {Config}
+ */
+function getConfigFromOptions(options) {
+  if (!(options instanceof Config)) options = loadConfig(options.id, options);
+  assert(options.str('id'), 'must pass an id to test config options');
+  // making a copy from options and data properties
+  // to avoid any mutations
+  return loadConfig(options.str('id'), {
+    ...options.options,
+    ...options.data,
+  });
+}
+
+/*
+ * Retrieve a config from clients directory
+ * @param {string} id - id of client to retrieve
+ * @returns {bcfg.Config} - bcfg object of config
+ */
+function getConfig(id) {
+  assert(typeof id === 'string', 'Client config must have an id');
+
+  const appConfig = loadConfig('bpanel');
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+  const config = loadConfig(id, { id, prefix: clientsPath });
+
+  const path = resolve(clientsPath, `${id}.conf`);
+  if (!fs.existsSync(path)) {
+    const error = new Error(`File ${id}.conf not found`);
+    error.code = 'ENOENT';
+    throw error;
+  }
+
+  config.open(`${id}.conf`);
+  return config;
+}
+
+/*
+ * create client config
+ * Note: This will actually create the file in your bpanel prefix location
+ * @param {string} id - id for the client
+ * @param {Object} options object for a bcoin/hsd client
+ * @param {bool} force - whether or not to force config creation if client
+ * can't connect
+ * @returns {bcfg.Config}
+ */
+async function createClientConfig(id, options = {}, logger) {
+  assert(typeof id === 'string', 'Must pass an id as first paramater');
+  assert(logger, 'Expected logger to be passed');
+
+  let clientConfig = options;
+  if (!(options instanceof Config))
+    clientConfig = getConfigFromOptions({ id, ...options });
+
+  const appConfig = loadConfig('bpanel', options);
+  const clientsDir = appConfig.str('clients-dir', 'clients');
+
+  // get full path to client configs relative to the project
+  // prefix which defaults to `~/.bpanel`
+  const clientsPath = resolve(appConfig.prefix, clientsDir);
+
+  let configTxt = '';
+  for (let key in clientConfig.options) {
+    const configKey = key
+      .replace('-', '')
+      .replace('_', '')
+      .toLowerCase();
+    const text = `${configKey}: ${clientConfig.options[key]}\n`;
+    configTxt = configTxt.concat(text);
+  }
+  if (!fs.existsSync(clientsPath)) {
+    logger.warning(
+      'Could not find requested client directory at %s. Creating new one...',
+      clientsPath
+    );
+    fs.mkdirpSync(clientsPath);
+  }
+
+  const configPath = `${clientsPath}/${clientConfig.str('id')}.conf`;
+  logger.debug('Adding config to path: %s', configPath);
+  fs.writeFileSync(configPath, configTxt);
+  return clientConfig;
+}
+
+/*
+ * Simple utility that deletes a config
+ * Does not throw errors if client is not found
+ * NOTE: This does not confirm. Deletion is final!
```

Comment:
Not sure how relevant this is because the front end could have a pop up confirmation if it wanted to

Reply:
yeah, and that's how it's implemented. I guess I just wanted to document that if this endpoint is hit and the config exists, the deletion happens. Basically it documents that this should be handled by the client.